pandas/_libs/lib.pyx

```diff
@@ -2802,7 +2803,8 @@ class NoDefault(Enum):
 
 
 # Note: no_default is exported to the public API in pandas.api.extensions
-no_default = NoDefault.no_default  # Sentinel indicating the default value.
+no_default = _NoDefault.no_default  # Sentinel indicating the default value.
+NoDefault = Literal[_NoDefault.no_default]
```

Comment:
Instead of defining `NoDefault` in lib.pyx and lib.pyi it could be defined in _typing.py (would require changing a lot of import statements)

Reply:
its actually better where it is as we avoid circular deps this way